src/unix.rs

```diff
@@ -111,16 +120,23 @@ impl Client {
             FromEnvErrorInner::CannotParse("expected a path after `fifo:`".to_string())
         })?;
         let path = Path::new(path_str);
-        let file = OpenOptions::new()
-            .read(true)
-            .write(true)
-            .open(path)
-            .map_err(|err| FromEnvErrorInner::CannotOpenPath(path_str.to_string(), err))?;
-
-        Ok(Some(Client::Fifo {
-            file,
-            path: path.into(),
-            is_non_blocking: AtomicBool::new(false),
+
+        let open_file = || {
+            // Open with read write is necessary, since open with read-only or
+            // write-only could block the thread until another thread open it
+            // with write-only or read-only (or RDWR) correspondingly.
```

Comment:
```suggestion
            // Open with read write is necessary, since opening with read-only or
            // write-only could block the thread until another thread opens it
            // with write-only or read-only (or RDWR) correspondingly.
```